src/lib/insights/client.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  Hits,
+  InsightsClient,
+  InsightsClientMethod,
+  InsightsClientPayload,
+  Renderer,
+  RenderOptions,
+  Unmounter,
+  WidgetFactory,
+} from '../../types';
+import find from 'lodash/find';
+
+export const inferPayload = ({
+  method,
+  results,
+  hits,
+  objectIDs,
+}: {
+  method: InsightsClientMethod;
+  results: RenderOptions['results'];
+  hits: Hits<unknown>;
+  objectIDs: string[];
+}): InsightsClientPayload => {
+  const { index, queryID } = results;
+  switch (method) {
+    case 'clickedObjectIDsAfterSearch': {
+      const positions = objectIDs.map(objectID => {
+        const hit = find(hits, hit => hit.objectID === objectID);
+        if (typeof hit === 'undefined') {
+          throw new Error(
+            `Could not find ObjectID ${objectID} in the returned hits.`
```

Comment:
Add more context mentioning the method `clickedObjectIDsAfterSearch`.

Reply:
ok, replaced with: 
> `Could not find ObjectID ${objectID} passed to 'clickedObjectIDsAfterSearch' in the returned hits. This is necessary to infer absolute position.`